src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -658,5 +690,29 @@ private static bool TryParseIdRangeEntry(string entry, out string prefix, out in
 
             return prefix.Length > 0 && suffixStr.Length > 0 && int.TryParse(suffixStr, out suffix);
         }
+
+        private static bool IsReservedDiagnosticId(string ruleId, string assemblyName)
+        {
+            if (ruleId.Length < 6)
```

Comment:
Do we want to flag cases like `CA123`?

Reply:
I think we want to flag even these cases - emphasis is on avoiding non-Microsoft packages having analyzers with CA, BC and CS prefixes.